util/math.h

```diff
@@ -6,11 +6,13 @@
 #pragma once
 
 #include <assert.h>
-#include <stdint.h>
 #ifdef _MSC_VER
 #include <intrin.h>
 #endif
 
+#include <cstdint>
+#include <type_traits>
+
```

Comment:
Seems these includes are not needed.

Reply:
I just assume some compiler will want them:

https://en.cppreference.com/w/cpp/types/is_integral
https://en.cppreference.com/w/cpp/types/integer